File: front/src/components/authentication/dtos/user-account-profile-update-dto.ts
import { AddressDto } from "shared/components/address/dtos/address-dto";
import { ContactDto } from "shared/components/contact/dtos/contact-dto";
import { RootBase } from "shared/entities-dtos/root-base";

export class UserAccountProfileUpdateDto extends RootBase {
  address!: AddressDto;
  contact!: ContactDto;
}

File: back/Application/Auth/Dtos/UserAccountProfileUpdateDto.cs
// using Application.Shared.Dtos;
// using Domain.Entities.System.Profiles;

// namespace Application.Auth.Dtos;

// public class UserAccountProfileUpdateDto : RootBaseDto
// {
//     public  string? UserAccountId { get; set; }
//     public int BusinessProfileId { get; set; }
//     public required AddressDto Address { get; set; }
//     public required ContactDto Contact { get; set; }
// }

// public static class UserAccountProfileUpdateDtoMapper
// {
//     public static UserProfile ToUpdate(this UserAccountProfileUpdateDto dto, UserProfile db)
//     {

//         db.Id = dto.Id;
//         db.Deleted = dto.Deleted;
//         db.Registered = dto.Registered;
//         db.BusinessProfileId = dto.BusinessProfileId;
//         db.UserAccountId = dto.UserAccountId;
//         db.Address = dto.Address;
//         db.Contact = dto.Contact;
//         return db;
//     }


// }